src/status_im/ui/screens/wallet/subs.cljs

```diff
@@ -66,32 +66,46 @@
 
 (reg-sub :wallet/transactions
   (fn [db]
-    (get-in db [:wallet :transactions])))
-
-(defn filter-transactions [type transactions]
-  (filter #(= (:type %) type) transactions))
+    (group-by :type (get-in db [:wallet :transactions]))))
 
 (reg-sub :wallet/unsigned-transactions
   :<- [:wallet/transactions]
   (fn [transactions]
-    (filter-transactions :unsigned transactions)))
+    (:unsigned transactions)))
+
+(reg-sub :wallet/postponed-transactions
+  :<- [:wallet/transactions]
+  (fn [transactions]
+    (:postponed transactions)))
+
+(reg-sub :wallet/pending-transactions
+  :<- [:wallet/transactions]
+  (fn [transactions]
+    (:pending transactions)))
+
+(reg-sub :wallet/completed-transactions
+  :<- [:wallet/transactions]
+  (fn [{:keys [inbound outbound]}]
+    (->> (into inbound outbound)
+         (sort-by :timestamp))))
 
 (defn mini-str-date->keyword [mini-str-date]
   (keyword (str "sent-" (string/replace mini-str-date #" " "-"))))
 
 (reg-sub :wallet/transactions-history-list
-  :<- [:wallet/transactions]
-  (fn [transactions]
-    (let [{:keys [postponed pending inbound outbound]} (group-by :type transactions)
-          transaction-history-list                     [{:title "Postponed"
-                                                         :key :postponed
-                                                         :data (or postponed [])}
-                                                        {:title "Pending"
-                                                         :key :pending
-                                                         :data (or pending [])}]
-          completed-transactions                       (->> (into inbound outbound)
-                                                            (group-by #(datetime/date->mini-str-date (:timestamp %)))
-                                                            (map (fn [[k v]] {:title k
-                                                                              :key (mini-str-date->keyword k)
-                                                                              :data v})))]
+  :<- [:wallet/postponed-transactions]
```

Comment:
Can't we do this with a single subscription?

Reply:
I think it is better that way because  if you have an unsigned transaction that is added to the list of transactions this won't recompute this view as it doesn't depend on it

Reply:
Is there a reason why `unsigned` transactions should be merged with others?

Reply:
no it is just a design decision. I don't see a need for separation as it can be done with these kind of derived subscriptions.

Reply:
Those have a cost (performance and cognitive) so unless they are mandatory we don't want to introduce new ones.

Separating `unsigned` from the others has some sense has they are logically different and are not displayed together.